enarx-keep-sev-shim/asm/rcrt1.c

```diff
@@ -0,0 +1,48 @@
+// SPDX-License-Identifier: MIT
+// Taken from rcrt1.c musl libc
+// Only used as a template to generate code for `entry_ram64.S`
+#include <stddef.h>
+
+#define R_X86_64_RELATIVE   8
+
+#define REL_RELATIVE    R_X86_64_RELATIVE
+
+#define DYN_CNT 32
+#define DT_REL            17
+#define DT_RELSZ  18
+#define DT_RELA           7
+#define DT_RELASZ 8
+
+#define R_TYPE(x) ((x)&0x7fffffff)
+#define R_SYM(x) ((x)>>32)
+
+#define IS_RELATIVE(x,s) (R_TYPE(x) == REL_RELATIVE)
+
+
+void _start_c(size_t *arg1, size_t base, size_t *dynv)
+{
+    size_t i, dyn[DYN_CNT];
+    size_t *rel;
+    size_t rel_size;
+
+    for (i=0; i<DYN_CNT; i++) dyn[i] = 0;
+    for (i=0; dynv[i]; i+=2) if (dynv[i]<DYN_CNT)
+            dyn[dynv[i]] = dynv[i+1];
+
+    rel = (void *)(base+dyn[DT_REL]);
+    rel_size = dyn[DT_RELSZ];
+    for (; rel_size; rel+=2, rel_size-=2*sizeof(size_t)) {
+        if (!IS_RELATIVE(rel[1], 0)) continue;
+        size_t *rel_addr = (void *)(base + rel[0]);
+        *rel_addr += base;
+    }
+
+    rel = (void *)(base+dyn[DT_RELA]);
+    rel_size = dyn[DT_RELASZ];
+    for (; rel_size; rel+=3, rel_size-=3*sizeof(size_t)) {
+        if (!IS_RELATIVE(rel[1], 0)) continue;
+        size_t *rel_addr = (void *)(base + rel[0]);
+        *rel_addr = base + rel[2];
+    }
+}
```

Comment:
If we're using C to generate assembly to copy/paste into another assembly file, why can't we just use Rust?

Reply:
I might do this in "C" or Rust, but I think I need to correct the address of the function to call first with it... so chicken and egg.

Reply:
Let me try this.